src/client/Microsoft.Identity.Client/Internal/ClientCredential/SignedAssertionDelegateClientCredential.cs

```diff
@@ -36,17 +37,41 @@ public async Task AddConfidentialClientParametersAsync(
             string tokenEndpoint,
             CancellationToken cancellationToken)
         {
-            string signedAssertion = await (_signedAssertionDelegate != null 
-                ? _signedAssertionDelegate(cancellationToken).ConfigureAwait(false)
-                : _signedAssertionWithInfoDelegate(new AssertionRequestOptions {
+            if (_signedAssertionDelegate != null)
+            {
+                // If no "AssertionRequestOptions" delegate is supplied
+                string signedAssertion = await _signedAssertionDelegate(cancellationToken).ConfigureAwait(false);
+                oAuth2Client.AddBodyParameter(OAuth2Parameter.ClientAssertionType, OAuth2AssertionType.JwtBearer);
+                oAuth2Client.AddBodyParameter(OAuth2Parameter.ClientAssertion, signedAssertion);
+            }
+            else
+            {
+                // Build the AssertionRequestOptions and conditionally set ClientCapabilities
+                var assertionOptions = new AssertionRequestOptions
+                {
                     CancellationToken = cancellationToken,
                     ClientID = requestParameters.AppConfig.ClientId,
                     TokenEndpoint = tokenEndpoint
-                }).ConfigureAwait(false));
+                };
 
-            oAuth2Client.AddBodyParameter(OAuth2Parameter.ClientAssertionType, OAuth2AssertionType.JwtBearer);
-            oAuth2Client.AddBodyParameter(OAuth2Parameter.ClientAssertion, signedAssertion);
-        }
+                // Only set client capabilities if they exist and are not empty
+                var configuredCapabilities = requestParameters
+                    .RequestContext
+                    .ServiceBundle
+                    .Config
+                    .ClientCapabilities;
+
+                if (configuredCapabilities != null && configuredCapabilities.Any())
+                {
+                    assertionOptions.ClientCapabilities = configuredCapabilities;
```

Comment:
You are still missing the Claims.

Reply:
this PR is only add capability as a feature @bgavrilMS. Claims was already shipped in MSAL. Missing claims propagation should be tracked as a bug fix and be a separate PR.

Reply:
Tracking  missing claims using https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/5143